### ISSUE TITLE
Memory Use Monitoring Helps

### ISSUE THREAD

**Issue by 21771** [Bears-R-Us/arkouda#2056]
**Feature Description**
Please provide a clear and concise description of the feature you are requesting.

1) Create user resettable variable containing max memory bytes used 
- Not requesting additional memory use checks, but when memory use is checked then update this variable.

2) (Less important) Create user resettable max memory projected variable 
- When overMemLimits are calculated, keep a running max.

3) Include "percent of total memory" used as well as "bytes of memory used" in log lines.
- This feature involves logs whereas the two previous are accessible from client space.
- No additional checks requested, just include percentage on existing lines that output memory bytes
- percent is requested to facilitate visual checks (which aren't needed in the previous requests)

**Example Behavior**
If applicable, please provide a code block showing expected behavior of the feature.
N/A

**Describe Workarounds**
Is there currently a way to achieve the desired functionality (even if it is not ideal)? If so, please provide it for context.
N/A

**Additional context**
Add any other context or screenshots about the feature request here. If the requested feature is based on another library (i.e. numpy, pandas, scipy), please provide a link to their supporting documentation for reference.

Rationale: Resources are dedicated at invocation time of the arkouda server. With knowledge of resource use for existing jobs, it would be helpful to refine resource requests for subsequent jobs.

**Comment by 21771** [comment]
See parts of #3 in v2023.01.11 but the values might not reflect the correct percentages.

**Comment by hokiegeek2** [comment]
@21771 argh, sorry, the 821% was due to a bug that is fixed in Arkouda [v2023.04.07](https://github.com/Bears-R-Us/arkouda/releases/tag/v2023.04.07)

**Comment by 21771** [comment]
Oh ... I hadn't realized our default was behind. Thanks. Sorry to bring this up since it's already been addressed.

BTW, I was just going to submit a couple of issues related to registering DataFrames containing segarrays. If this area has been addressed since 03.01 then I'll wait until we update to note the issues. Please let me know if you want them entered. Easily tested with 1) registering df with segarrays, clearing memory and then reattaching, or 2) registering DF without segarrays and then registering one with.  Thanks again for the memory utilization info!

**Comment by hokiegeek2** [comment]
@21771 not at all, I apologize for the initial bug. You may want to check out the [v2023.03.24](https://github.com/Bears-R-Us/arkouda/releases/tag/v2023.03.24) and [v2023.04.07](https://github.com/Bears-R-Us/arkouda/releases/tag/v2023.04.07) release notes to see if the DataFrame-segarray registration issues you mentioned were addressed in either of those releases.

Tagging @pierce314159 @joshmarshall1 to look into these issues as they are more familiar w/ that part of the codebase

**Comment by pierce314159** [comment]
@21771 off hand i don't think anything in the latest release should've fixed bugs with registering dataframes which contain segarrays. So it would be great if you could go ahead and post an issue/reproducer for that!
